Source: Cameron Nemo <cnemo@tutanota.com>
Upstream: false

Add in the binfmts for ppcle and cris architectures.

--- scripts/qemu-binfmt-conf.sh.orig	2021-04-27 16:04:03.556284793 -0700
+++ scripts/qemu-binfmt-conf.sh	2021-04-27 16:05:10.906096654 -0700
@@ -2,9 +2,9 @@
 # Enable automatic program execution by the kernel.
 
 qemu_target_list="i386 i486 alpha arm armeb sparc sparc32plus sparc64 \
-ppc ppc64 ppc64le m68k mips mipsel mipsn32 mipsn32el mips64 mips64el \
+ppc ppc64 ppcle ppc64le m68k mips mipsel mipsn32 mipsn32el mips64 mips64el \
 sh4 sh4eb s390x aarch64 aarch64_be hppa riscv32 riscv64 xtensa xtensaeb \
-microblaze microblazeel or1k x86_64"
+microblaze microblazeel or1k x86_64 cris"
 
 i386_magic='\x7fELF\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x03\x00'
 i386_mask='\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff'
@@ -50,6 +50,10 @@
 ppc64_mask='\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff'
 ppc64_family=ppc
 
+ppcle_magic='\x7f\x45\x4c\x46\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x14\x00'
+ppcle_mask='\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\x00'
+ppcle_family=ppcle
+
 ppc64le_magic='\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x15\x00'
 ppc64le_mask='\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\x00'
 ppc64le_family=ppcle
@@ -136,6 +140,10 @@
 or1k_mask='\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff'
 or1k_family=or1k
 
+cris_magic='\x7f\x45\x4c\x46\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x4c\x00'
+cris_mask='\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff'
+cris_family=cris
+
 qemu_get_family() {
     cpu=${HOST_ARCH:-$(uname -m)}
     case "$cpu" in
@@ -148,7 +156,7 @@
     "Power Macintosh"|ppc64|powerpc|ppc)
         echo "ppc"
         ;;
-    ppc64el|ppc64le)
+    ppc64el|ppc64le|ppcle)
         echo "ppcle"
         ;;
     arm|armel|armhf|arm64|armv[4-9]*l|aarch64)
@@ -163,6 +171,9 @@
     riscv*)
         echo "riscv"
         ;;
+    cris*)
+        echo "cris"
+        ;;
     *)
         echo "$cpu"
         ;;
